.github/workflows/lint-rst.yaml

```diff
@@ -0,0 +1,82 @@
+name: RST checks
+
+# Any change in triggers needs to be reflected in the concurrency group.
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '**.rst'
+  workflow_dispatch:
```

Comment:
We probably want to run this on `pull_request` event (and skip if no rst files are changed) instead of triggering this manually, right?

Reply:
> Edit: sorry, I misread your initial PR comment. I still think we should run this on PRs rather than on stable branch pushes.

Ok. I can modify the action to be triggered on PRs rather than on pushes. It will probably demand a change in the git command.

Concerning the `workflow_dispatch` option, would I keep it?